clusterstate/clusterstate.go

```diff
@@ -51,6 +51,28 @@ func New(config Config) (*ClusterState, error) {
 func (c *ClusterState) Test(ctx context.Context) error {
 	var err error
 
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "getting current AZs in tenant cluster")
+		zones, err := c.provider.GetClusterAZs(ctx)
```

Comment:
AWS legacy runs these tests - does this break them?

Reply:
Yes since those methods wouldn't be implemented. I could implement them. What would be a good way to fetch the current AZs being used by the cluster using the AWS client? Are the instances tagged with the AZ's? any easy way?

Reply:
Something like this? https://github.com/giantswarm/aws-operator/pull/2134

Reply:
Maybe someone of @giantswarm/team-firecracker can take a look - I am not sure currently how this is supposed to be.

Reply:
The idea with the abstracted e2e tests here was to have a scaffold that produces a scenario. The assertion was meant to be done in the test running the scenario. So I think `ExpectedAZs` is wrong for the design aspirations we had so far. When we go down that route people start to add all kinds of crap in here and the whole thing explodes again. I would like to prevent that and instead keep it nice and clean as it is right now. A provider may not even have AZ features the test wants to verify, like KVM. So I think the AZ checks should be implemented in the provider operator and the e2e test scenario here should stay as it is. My two cents.